### DISCUSSION TITLE
Some suggestions for this software

### DISCUSSION THREAD

**Discussion by SMFDrummer** [eythaann/Seelen-UI#200] [Ideas]
Since the software was released, the feedback from all users who have installed and used it on any platform is mostly negative, such as: if it is set arbitrarily, the small triangle on the toolbar cannot be clicked and keeps flashing, it should trigger a bug and loop on its own. It is already version 2.0.3 and the most basic stability of the software has not been resolved. The frequency of bug triggers is even more severe than MyDockFinder. If developers still cannot effectively solve various bugs that even affect normal use, it is recommended to release all projects to pre release and indicate the serious problems that may be caused by user use. You can also write down all the effective solutions for the issues raised in a markdown file for users to refer to. If it still cannot effectively solve various bug problems and blindly innovate, then I suggest developers slow down their development speed. If all users are attracted by your appearance and uninstall your software due to serious bugs in the program, how many people will pay attention to and use your software in the end?

**Comment by eythaann** [comment]
yep, you have a big point on this, but, just wait. I'm currently alone on this project so I can't support all the stuff that a big project needs only by myself. Also the project is free so I'm not getting funds for the app to start getting more people into the project.

Now, growth just happens I wasn't planning on promoting the app yet, just blogs start posting about the app.

btw, this is normal when fasting growing, if nothing is on fire, you're not moving fast enough.

**Reply by mchao123** [reply]
Perhaps consider splitting the project, with the main project providing only the api, so that users can mount their own desired taskbars, and more magical features.Provide externally exposed interfaces and documentation so that more developers can participate in the development of the project's features.For example, I personally do not like the status quo taskbar shadows and margins, if I want to change can only recompile the project, perhaps the front-end part can be split out, so I only need to modify the taskbar part.This is just an idea I had, thank you for open sourcing this project.

**Reply by eythaann** [reply]
@mchao123 you can change the style of things using css themes, btw I will be enhancing the development ecosystem, a library is already done but not exported to be used yet, this is planned for future versions with the widget system where all the API will be accessible via custom widgets

**Reply by mchao123** [reply]
Thanks for the reply, I was just kind of assuming, but since this is already planned, I'm very much looking forward to it.I can help in any way I can, but I'm only good with vue3. Would it be possible to think about having the widget provide its own settings page, providing a centralized mounting and unmounting, like the vscode extension.That way maybe the taskbar could also be used as a widget of sorts.

**Reply by eythaann** [reply]
native modules are/will be widgets basically, so yeah each widget will be able to declare settings and this will be shown on the settings window, with a specific tab for each widget ;) also the incoming Plugins system are "expansions" for specific a widget so plugins for toolbar will be different than plugins for dock or plugins for any other custom widget.

**Reply by eythaann** [reply]
the widgets are library agnostic, the only requirement will be the bundled js, html and css, so you can create the widget on any library including any language like python and later just compile it to JS.
